### PR TITLE
Ability to replace a node with another node

### LINKED ISSUE #115: Ability to replace nodes with text and/or other nodes
Oga doesn't have a dedicated method for replacing one XML node with another, instead you have to use the following pattern:

```ruby
some_node.before(some_other_node)
some_node.remove
```

It would be nicer if this could instead be written as following:

```ruby
some_node.replace(some_other_node)
```

This would basically just use the above pattern but hide it from the end user. Nokogiri also has a `replace` method which if I'm not mistaken can use both an XML node and a String. I'm not 100% sure if I like the idea of also supporting a String, but at the very least an XML node should be supported.

### PR DESCRIPTION
**Pull request by ttasanen**
Really  simple implementation of `#replace`. What do you think?
I also added "duplicate" test case for `Oga::XML::Text` do you think this is necessary?

closes #115

### CODE REVIEW

**Comment by YorickPeterse** [comment]
In the initial ticket what I meant with supporting text was this:

    some_node.replace('this is some sample text')

This would basically translate to the following:

    text_node = Oga::XML::Text.new(:text => 'this is some sample text')
    some_node.replace(text_node)

The thing here is that I'm not sure yet if I like the idea of `replace` taking either a `XML::Node` _or_ a String, though at the same time there's not really anything wrong with it either.

Other than that this PR looks fine so far.

**Comment by ttasanen** [comment]
Ok, I misunderstood the text part.

But yeah, I don't generally like the idea of a method accepting either `XML::Node` or `String`. This usually just adds unnecessary conditionals to a method. Like in methods that Rails is full of. eg.

```ruby
def foo(id_or_hash_of_options, *args)
  if id_or_hash_of_options.is_a?(Hash)
    foo_with_hash(id_or_hash_of_options, *args)
  else
    foo_with_id(id_or_hash_of_options, *args)
  end
end
``` 
But on the other hand, this makes the API simpler for the user in some cases. Don't know if this is one. IMO it's always better to methods to do explicitly do one thing.

Should there be two methods here? `replace` and `replace_text` or something similar?

**Comment by YorickPeterse** [comment]
Thinking of it the code wouldn't be too difficult, something like:

```ruby
def replace(other)
  if other.is_a?(String)
    other = Text.new(:text => other)
  end

  before(other)
  remove
end
```

Thinking of it I don't really have an issue with this code so it's fine to include in the PR.

**Comment by ttasanen** [comment]
Yeah, I updated the PR with the changes.

File: spec/oga/xml/node_spec.rb [hunk 1]
```diff
@@ -179,6 +179,29 @@
...
+
+      @container.children[0].should == other
+    end
+
+    it 'replaces a node with Oga::XML::Text when called with String' do
+      @node.replace('a string')
+
+      @container.children[0].should be_an_instance_of(Oga::XML::Text)
+      @container.children[0].text.should == 'a string'
+
```
Comment (YorickPeterse) [inline spec/oga/xml/node_spec.rb]: Super picky: this empty newline isn't needed.

Comment (ttasanen) [inline spec/oga/xml/node_spec.rb]: haha, good that someone is strict about the code consistency.. Fixed and squashed the commits.

**Comment by YorickPeterse** [comment]
I'll wait for appveyor to finish running and then this should be good to go. Thanks for taking care of this!

**Pull request merged**